File: POO/tad Lista/c++/queue.cpp
#include <iostream>
#include "queue.h"

void Queue::enqueue(int value)
{
    list.insert(value); // Inserção no final da lista
}

void Queue::dequeue()
{
    if (list.getSize() > 0)
    {
        list.removeFirst(); // Remoção do início da lista (FIFO)
    }
    else
    {
        std::cout << "Fila Vazia!" << std::endl;
    }
}

int Queue::size()
{
    return list.getSize();
}

int Queue::front()
{
    return list.firstItem(); // Retorna o elemento no início da lista (FIFO)
}

int Queue::back()
{
    return list.lastItem(); // Retorna o elemento no início da lista (FIFO)
}

File: POO/tad Lista/java/Queue.java
public class Queue {
    private List list = new List();

    public void enqueue(int value) {
        list.insert(value);
    }

    public void dequeue() {
        if (list.getSize() > 0) {
            list.removeFirst();
        } else {
            System.out.println("Fila Vazia!");
        }
    }

    public int size() {
        return list.getSize();
    }

    public int front() {
        return list.firstItem();
    }

    public int back() {
        return list.lastItem();
    }
}
